### PR TITLE
[DO NOT MERGE] Make the comment header cell accessible

### LINKED ISSUE #1243: VoiceOver should select all of comment header except "..."
Don't need to select image view, username, and date label separately. Squish into one AX view (maybe using custom rect?) and make it readable:

"rnystrom commented 3 minutes ago"

### PR DESCRIPTION
**Pull request by BasThomas**
Fixes #1243.

I couldn't find out how to make it so the `...` is also accessible, but wanted to PR anyway just for looks. @Sherlouk, any idea maybe?

### CODE REVIEW

**Comment by Sherlouk** [comment]
Not sure how the view hierarchy is laid out, if it's nested in the same container view then we'd probably need to take it out of the container but keep constraints as they are - should keep it in the same location, but allow it to be essentially just on top thus allowed to use it

Also need to be careful to not block tapping on the header to open user profile

**Comment by BasThomas** [comment]
> Also need to be careful to not block tapping on the header to open user profile

What do you mean with this one?

And yeah, probably. I thought it would just surface as the `...` is the topmost subview and is a button, but as far as I can tell (haven't tested on device), it's not discoverable.

**Comment by Sherlouk** [comment]
Once a view is marked as an accessibility element, subviews are not discoverable either by tapping or by swiping through elements 😢 

I mean that if you're in VoiceOver, and I double tap on the header (select) then it should open the user's profile

**Comment by BasThomas** [comment]
> I mean that if you're in VoiceOver, and I double tap on the header (select) then it should open the user's profile

https://github.com/rnystrom/GitHawk/pull/1397/files#diff-f852a51ab57af4d90983098a74165159R128 :)

**Comment by Sherlouk** [comment]
ahh misread what that was doing my bad

**Comment by BasThomas** [comment]
Tell me! How can I improve it?

File: Classes/Issues/Comments/Details/IssueCommentDetailCell.swift [hunk 1]
```diff
@@ -127,19 +147,25 @@ final class IssueCommentDetailCell: IssueCommentBaseCell, ListBindable {
...
+            editedLabel.text = "\(Constants.Strings.bullet) \(editedString)"
 
             let detailFormat = NSLocalizedString("%@ edited this issue %@", comment: "")
             editedLabel.detailText = String(format: detailFormat, arguments: [editedLogin, editedDate.agoString])
         } else {
             editedLabel.isHidden = true
+            editedString = ""
         }
+
+        accessibilityLabel = NSLocalizedString("\(viewModel.login) commented \(viewModel.date.agoString). \(editedString)", comment: "")
```
Comment (Sherlouk) [inline Classes/Issues/Comments/Details/IssueCommentDetailCell.swift]: Not sure using a localized string is best here, there's no way we could translate every possible mutation of this -- Maybe if we use `String(format:` where the format is localizable?

File: Classes/Issues/Comments/Details/IssueCommentDetailCell.swift [hunk 2]
```diff
@@ -114,6 +125,15 @@ final class IssueCommentDetailCell: IssueCommentBaseCell, ListBindable {
         delegate?.didTapProfile(cell: self)
     }
 
+    @objc func onTapCell() {
+        // Only enable the tap of the cell in VoiceOver;
+        // otherwise the avatar and label work as advertised.
+        // It seems like we need to add a recognizer to the cell
+        // to enable an action while using VoiceOver.
+        guard UIAccessibilityIsVoiceOverRunning() else { return }
+        delegate?.didTapProfile(cell: self)
```
Comment (Sherlouk) [inline Classes/Issues/Comments/Details/IssueCommentDetailCell.swift]: Personal preference for me but when it's such a trivial function like this I think it reads better to use an `if`. I personally feel it reads a bit easier!

File: Classes/Issues/Comments/Details/IssueCommentDetailCell.swift [hunk 3]
```diff
@@ -83,6 +93,7 @@ final class IssueCommentDetailCell: IssueCommentBaseCell, ListBindable {
 
         editedLabel.font = Styles.Fonts.secondary
         editedLabel.textColor = Styles.Colors.Gray.light.color
+        editedLabel.isAccessibilityElement = false
```
Comment (Sherlouk) [inline Classes/Issues/Comments/Details/IssueCommentDetailCell.swift]: Do we need to set everything to false if the parent is set to true?

File: Classes/Issues/Comments/Details/IssueCommentDetailCell.swift [hunk 4]
```diff
@@ -49,9 +55,10 @@ final class IssueCommentDetailCell: IssueCommentBaseCell, ListBindable {
         loginLabel.font = Styles.Fonts.title
         loginLabel.textColor = Styles.Colors.Gray.dark.color
         loginLabel.isUserInteractionEnabled = true
+        loginLabel.isAccessibilityElement = false
         loginLabel.addGestureRecognizer(UITapGestureRecognizer(
             target: self,
-            action: #selector(IssueCommentDetailCell.onTapLoginLabel))
+            action: #selector(onTapLoginLabel))
```
Comment (Sherlouk) [inline Classes/Issues/Comments/Details/IssueCommentDetailCell.swift]: 👍

File: Classes/Issues/Comments/Details/IssueCommentDetailCell.swift (hunk 3 above)
Comment (BasThomas) [reply Classes/Issues/Comments/Details/IssueCommentDetailCell.swift]: Don't think so no, we can change this.

File: Classes/Issues/Comments/Details/IssueCommentDetailCell.swift (hunk 2 above)
Comment (BasThomas) [reply Classes/Issues/Comments/Details/IssueCommentDetailCell.swift]: I don't mind; feel free to change it

File: Classes/Issues/Comments/Details/IssueCommentDetailCell.swift (hunk 1 above)
Comment (BasThomas) [reply Classes/Issues/Comments/Details/IssueCommentDetailCell.swift]: Oops, of course.

**Comment by rnystrom** [comment]
Cleaning up

**Comment by BasThomas** [comment]
Yeah I’ll take this on again soon. 

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Pull request closed without merging**